test/core/networking/protocol/authentication_test.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:test/test.dart';
+
+import 'package:flyde/core/networking/protocol/authentication.dart';
+import 'package:flyde/core/networking/protocol/protocol.dart';
+
+void main() {
+  group('AuthRequest', () {
+    test('In ProtocolDelegate.elements', () {
+      expect(ProtocolDelegate.elements['AuthRequest'], equals(isA<Deserialize>()));
+    });
+
+    test('Can serialize', () {
+      final AuthRequest request = AuthRequest(username: "testUser", password: "testPassword");
+
+      final String serialized = ProtocolDelegate.serialize(request);
+
+      expect(
+        serialized,
+        equals(
+          '{"type":"AuthRequest","data":{"username":"testUser","password":"testPassword"}}',
+        ),
+      );
+    });
+
+    test('Can deserialize', () {
+      final String request =
+          '{"type":"AuthRequest","data":{"username":"testUser","password":"testPassword"}}';
+
+      final dynamic deserialized = ProtocolDelegate.deSerialize(request);
+
+      expect(deserialized, equals(isA<AuthRequest>()));
```

Comment:
```suggestion
      expect(deserialized, isA<AuthRequest>());
```

Please check for similar issues